base/stream.jl

```diff
@@ -876,6 +894,14 @@ function stop_reading(stream::LibuvStream)
     end
 end
 
+function readbytes!(s::LibuvStream, b::AbstractArray{UInt8}, nb=length(b))
+    wait_readnb(s, nb)
+    nr = nb_available(s)
+    resize!(b, nr) # shrink to just contain input data if was resized
+    read!(s.buffer, b)
```

Comment:
it also doesn't support `read` or `write` (it uses `send` and `recv`), so i'm not too worried about that. we could extend the type-hierarchy to exclude it (because it isn't a stream) explicitly, but to what end?